### ISSUE TITLE
Is training FLUX with NF4 possible?

### ISSUE THREAD

**Issue by gesen2egee** [bghira/SimpleTuner#737]
I saw on Reddit that someone achieved significant memory savings and speed improvements using the NF4 version of FLUX on Forge. 
The quality impact wasn't very significant. 
[](https://www.reddit.com/r/StableDiffusion/s/f3WfHLMHml
)
I wanted to ask if introducing NF4 in training is feasible. 
Perhaps it could enable FLUX training on consumer-grade GPUs.

**Comment by bghira** [comment]
we rely in third party libraries to do thie quantisation because of how complex this gets while training. you will have to ask optimum-quanto devs that

**Comment by gesen2egee** [comment]
Hi, thank you for your reply. I would like to ask if it is possible to use `bitsandbytes` for quantization instead of `quanto`. `bitsandbytes` has already implemented NF4.

**Comment by bghira** [comment]
everything _else_ the trainer depends on has to support it too. these are core dependencies like Accelerate and PEFT and the Diffusers projects.

**Comment by gesen2egee** [comment]
I roughly understand what you mean. 
Is this related?
[x@RisingSayak](https://x.com/RisingSayak/status/1823544035111301266)

**Comment by bghira** [comment]
yes that is inference related. trust me if it worked here sayak would open a pull request

**Comment by bghira** [comment]
@sayakpaul is indeed working on this :)
